app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    previous_runs = {}
+
+    def inner(*args) -> object:
+        nonlocal previous_runs
```

Comment:
nonlocal is redundant here because you don't use an assignment operator with this variable and only call its methods.

Reply:
Nonlocal removed